data/src/main/scala/ch.epfl.scala.index.data/Main.scala

```diff
@@ -0,0 +1,132 @@
+  */
+object Main {
+
+  private val logger = LoggerFactory.getLogger("ch.epfl.scala.index.data")
+
+  /**
+    * Update data:
+    *  - pull the latest data from the 'contrib' repository
+    *  - download data from Bintray and update the ElasticSearch index
+    *  - commit the new state of the 'index' repository
+    *
+    * @param args 3 arguments:
+    *              - Name of a step to execute (or “all” to execute all the steps)
+    *              - Path of the 'contrib' Git repository
+    *              - Path of the 'index' Git repository
+    */
+  def main(args: Array[String]): Unit = {
+    implicit val system = ActorSystem()
+    import system.dispatcher
+    implicit val materializer = ActorMaterializer()
+
+    val pathFromArgs =
+      if (args.isEmpty) Nil
+      else args.toList.tail
+
+    val paths = DataPaths(pathFromArgs.take(2))
+    val bintray: LocalRepository = LocalRepository.Bintray
+
+    val steps = List(
+      // List POMs of Bintray
+      Step("list")({ () =>
+        val listPomsStep = new BintrayListPoms(paths)
+
+        // TODO: should be located in a config file
+        val versions = List("2.12", "2.11", "2.10")
+
+        for (version <- versions) {
+          listPomsStep.run(version)
+        }
+
+        /* do a search for non standard lib poms */
+        for (lib <- NonStandardLib.load(paths)) {
+          listPomsStep.run(lib.groupId, lib.artifactId)
+        }
+      }),
+
+      // Download POMs from Bintray
+      Step("download")(() => new BintrayDownloadPoms(paths).run()),
+
+      // Download parent POMs
+      Step("parent")(() => new DownloadParentPoms(bintray, paths).run()),
+
+      // Download additional information about projects from Github
+      Step("github")(() => new GithubDownload(paths).run()),
+
+      // Re-create the ElasticSearch index
+      Step("elastic")(() => new SeedElasticSearch(paths).run())
+    )
+
+    val stepsToRun =
+      args.headOption match {
+        case Some("all") => steps
+        case Some(name)  =>
+          steps.find(_.name == name)
+            .fold(sys.error(s"Unknown step: $name. Available steps are: ${steps.map(_.name).mkString(" ")}."))(List(_))
+        case None        => sys.error(s"No step to execute. Available steps are: ${steps.map(_.name).mkString(" ")}.")
+      }
+
+    inPath(paths.contrib) { sh =>
+      logger.info("Pulling the latest data from the 'contrib' repository")
+      sh.exec("git", "checkout", "master")
+      sh.exec("git", "remote", "update")
+      sh.exec("git", "pull", "origin", "master")
+    }
+
+    logger.info("Executing steps")
+    stepsToRun.foreach(_.run())
+
+    inPath(paths.index) { sh =>
+      logger.info("Saving the updated state to the 'index' repository")
+      sh.exec("git", "add", "-A")
+      def now() = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss"))
+      sh.exec("git", "commit", "--allow-empty", "-m", '"' +: now() :+ '"')
+    }
```

Comment:
Note that we don’t push our changes. That’s only to make development easier: we want only the actual server to push its `ìndex` content to the github scaladex-index repository.